devel/gpgme/Portfile

```diff
@@ -29,34 +29,15 @@ depends_build       port:pkgconfig
 
 depends_lib         port:pth \
                     port:libgpg-error \
-                    port:libassuan
+                    port:libassuan \
+                    path:bin/gpg2:gnupg-stable
 
 patchfiles          patch-configure.ac.diff \
-                    patch-fixup-GpgmeppConfig.cmake.diff
-
-variant gnupg20 conflicts gnupg21 description {Use GnuPG 2.0.x} {
-    depends_build-append    port:gnupg2
-    depends_run             port:gnupg2
-    patchfiles-append       patch-tests-gpg-Makefile.am.diff
-
-    post-patch {
-        reinplace "s|@GPG@|${prefix}/bin/gpg2|" ${worksrcpath}/src/gpgme-config.in
-    }
-}
-variant gnupg21 conflicts gnupg20 description {Use GnuPG 2.1.x} {
-    depends_build-append    port:gnupg21
-    depends_run             port:gnupg21
-
-    post-patch {
-        reinplace "s|@GPG@|${prefix}/bin/gpg|" ${worksrcpath}/src/gpgme-config.in
-    }
-}
-# default to gnupg20 for now for backwards compatibility
-if {![variant_isset gnupg20] && ![variant_isset gnupg21]} {
-    variant_set             gnupg20
-}
+                    patch-fixup-GpgmeppConfig.cmake.diff \
+                    patch-tests-gpg-Makefile.am.diff
 
 post-patch {
+    reinplace "s|@GPG@|${prefix}/bin/gpg2|" ${worksrcpath}/src/gpgme-config.in
```

Comment:
I have made a separate comment, but I think it would be more elegant if `bin/gpg` is always the stable version and the few ports that depend on legacy invoke it via `bin/gpg1`. It looks backwards to me to have the ones depending on the "mainstream" one having to invoke a non-default binary.

Reply:
For now, I want to keep it installing `gpg`. In about a year, I want to make it only install `gpg1` and have no other port depend on it, with it being co-installable with the other ports. That way, users that still need `gnupg1` can install it, but it's not pulled in by default by any other port. Essentially deprecation in a year.

For now, I'd like to keep them conflicting.

Reply:
The part that I don't quite follow is what is gained by doing that one year from now compared to now. Less work? Potential less breakage? What are the ports that would pull gpg1 using the default set of variants?

I'm sure you have a good reason, though. I'm just curious to find out what I'm missing.

Reply:
Merely to let people keep their setup (mostly, sans the agent and dirmngr support...) for a little bit longer if they are determined to. Those that chimed in with a `+1` already use the modern version, but those that want to stay on `gnupg1` would definitely give a `-1`.

Since we have to keep at least some stub ports around anyway, I find that a good balance between forcing users to the modern version and giving them a grace period.

I probably should add a notes message about that to the new `gnupg1` port, that I think of it, though.

Reply:
Note added.

Reply:
I couldn't find any ports that depend on gnupg, but maybe my text search isn't working right.

Reply:
Regarding making gnupg1 and gnupg2 / -stable co-installable: Won't you run into the problem that the keystore formats are not compatible?

Re port naming: I would keep gnupg2 as what you now named gnupg-stable, then you can in a year from now move that into gnupg and make gnupg2 obsolete. If necessary we can add a gnupg-devel port to be consistent with macports conventions, but we won't have a -stable port that we already know we have to deprecate in a year.

Reply:
Yes, keystore formats differ, but that's fine.

The co-installable `gnupg1` port will be provided as a courtesy, but not referenced within MacPorts any longer. Thus, it's only useful to decrypt old data, which is not possible with the modern GnuPG version (cases like these exist) and we assume that users either have an old keyring storage format backed up, or that they know how to export and import public and private keys (which can be used to exchange keys between the two formats.)

Yeah, calling it `gnupg2` would remove a stub-port now, but stub-ports are cheap. We don't maintain these, they just sit there for a year and get removed.

Anyway, I can rename it to `gnupg2`.